Feature/docker support

dbsectrainer opened the pull request:
- Add Dockerfile for containerization
- Add docker-compose.yml for easy deployment
- Add docker-entrypoint.sh for container initialization
- Add configuration templates and documentation
- Support API key configuration through environment variables
- Add comprehensive Docker setup instructions

Fixes #85

dbsectrainer commented:
@microsoft-github-policy-service agree

danmarshall commented:
Thanks! Does it also work on Windows? I see a .sh file

danmarshall commented:
I tried running, I got this error:
```sh
 => [data-formulator-dev development  4/12] WORKDIR /app                                                                                                                                  0.2s
 => [data-formulator-dev development  5/12] RUN mkdir -p /app/config                                                                                                                      0.5s
 => ERROR [data-formulator-dev development  6/12] COPY docker-entrypoint.sh /app/                                                                                                         0.0s
```

dbsectrainer commented on docker/Dockerfile:
```diff
@@ -0,0 +1,71 @@
+    && rm -rf /var/lib/apt/lists/*
+
+# Install Node.js and yarn
+RUN curl -fsSL https://deb.nodesource.com/setup_18.x | bash - \
```
Updating to v22

dbsectrainer commented:
> ```shell
>  => [data-formulator-dev development  4/12] WORKDIR /app                                                                                                                                  0.2s
>  => [data-formulator-dev development  5/12] RUN mkdir -p /app/config                                                                                                                      0.5s
>  => ERROR [data-formulator-dev development  6/12] COPY docker-entrypoint.sh /app/   
> ```

Addressed issue and recent change. successful build now.

Chenglong-MS commented:
I run into multiple issues trying to work with this docker setup:

1. I run into the problem of `Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "/app/docker-entrypoint.sh": stat /app/docker-entrypoint.sh: no such file or directory: unknown` doing docker compose. I need to fix by updating entrypoint setup in docker-compose.yml with `entrypoint: ["/bin/bash", "/app/docker/docker-entrypoint.sh"]`

2. Still have trouble opening the page on docker, I tried adding --host to do network mapping but it somehow doesn't work yet.

Chenglong-MS commented:
Thanks for the quick fix! I still run into some issue:

1. entrypoint: [ "/bin/bash", "/app/docker/docker-entrypoint.sh" ] --> entrypoint: [ "/bin/bash", "/app/docker-entrypoint.sh" ] since it is copied as `/app/docker-entrypoint.sh`

2. still have issue routing 127.0.0.1 to my browser

dbsectrainer commented:
> Thanks for the quick fix! I still run into some issue:
> 
> 1. entrypoint: [ "/bin/bash", "/app/docker/docker-entrypoint.sh" ] --> entrypoint: [ "/bin/bash", "/app/docker-entrypoint.sh" ] since it is copied as `/app/docker-entrypoint.sh`
> 2. still have issue routing 127.0.0.1 to my browser

You should now be able to run command  "docker compose -f docker/docker-compose.yml down && docker compose -f docker/docker-compose.yml up --build data-formulator-dev" and see tool on port :5173

Chenglong-MS commented:
Can you do this PR to dev branch? I’ll still need to tweak some minor stuff there, especially for the data-formulator-dev

________________________________
From: Donnivis Baker ***@***.***>
Sent: Monday, March 3, 2025 11:26:07 AM
To: microsoft/data-formulator ***@***.***>
Cc: Chenglong Wang ***@***.***>; Comment ***@***.***>
Subject: Re: [microsoft/data-formulator] Feature/docker support (PR #90)


Thanks for the quick fix! I still run into some issue:

  1.  entrypoint: [ "/bin/bash", "/app/docker/docker-entrypoint.sh" ] --> entrypoint: [ "/bin/bash", "/app/docker-entrypoint.sh" ] since it is copied as /app/docker-entrypoint.sh
  2.  still have issue routing 127.0.0.1 to my browser

You should now be able to run command "docker compose -f docker/docker-compose.yml down && docker compose -f docker/docker-compose.yml up --build data-formulator-dev" and see tool on port :5173

—
Reply to this email directly, view it on GitHub<https://github.com/microsoft/data-formulator/pull/90#issuecomment-2695342146>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AWJXEPEQUGQ4VY47DHKEWI32SSUE7AVCNFSM6AAAAABXDTTYCGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJVGM2DEMJUGY>.
You are receiving this because you commented.Message ID: ***@***.***>

[dbsectrainer]dbsectrainer left a comment (microsoft/data-formulator#90)<https://github.com/microsoft/data-formulator/pull/90#issuecomment-2695342146>

Thanks for the quick fix! I still run into some issue:

  1.  entrypoint: [ "/bin/bash", "/app/docker/docker-entrypoint.sh" ] --> entrypoint: [ "/bin/bash", "/app/docker-entrypoint.sh" ] since it is copied as /app/docker-entrypoint.sh
  2.  still have issue routing 127.0.0.1 to my browser

You should now be able to run command "docker compose -f docker/docker-compose.yml down && docker compose -f docker/docker-compose.yml up --build data-formulator-dev" and see tool on port :5173

—
Reply to this email directly, view it on GitHub<https://github.com/microsoft/data-formulator/pull/90#issuecomment-2695342146>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AWJXEPEQUGQ4VY47DHKEWI32SSUE7AVCNFSM6AAAAABXDTTYCGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJVGM2DEMJUGY>.
You are receiving this because you commented.Message ID: ***@***.***>

dbsectrainer commented:
Updated PR branch to dev as requested 

> Can you do this PR to dev branch? I’ll still need to tweak some minor stuff there, especially for the data-formulator-dev
> […](#)
> ________________________________ From: Donnivis Baker ***@***.***> Sent: Monday, March 3, 2025 11:26:07 AM To: microsoft/data-formulator ***@***.***> Cc: Chenglong Wang ***@***.***>; Comment ***@***.***> Subject: Re: [microsoft/data-formulator] Feature/docker support (PR #90) Thanks for the quick fix! I still run into some issue: 1. entrypoint: [ "/bin/bash", "/app/docker/docker-entrypoint.sh" ] --> entrypoint: [ "/bin/bash", "/app/docker-entrypoint.sh" ] since it is copied as /app/docker-entrypoint.sh 2. still have issue routing 127.0.0.1 to my browser You should now be able to run command "docker compose -f docker/docker-compose.yml down && docker compose -f docker/docker-compose.yml up --build data-formulator-dev" and see tool on port :5173 — Reply to this email directly, view it on GitHub<[#90 (comment)](https://github.com/microsoft/data-formulator/pull/90#issuecomment-2695342146)>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AWJXEPEQUGQ4VY47DHKEWI32SSUE7AVCNFSM6AAAAABXDTTYCGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJVGM2DEMJUGY>. You are receiving this because you commented.Message ID: ***@***.***> [dbsectrainer]dbsectrainer left a comment ([microsoft/data-formulator#90](https://github.com/microsoft/data-formulator/pull/90))<[#90 (comment)](https://github.com/microsoft/data-formulator/pull/90#issuecomment-2695342146)> Thanks for the quick fix! I still run into some issue: 1. entrypoint: [ "/bin/bash", "/app/docker/docker-entrypoint.sh" ] --> entrypoint: [ "/bin/bash", "/app/docker-entrypoint.sh" ] since it is copied as /app/docker-entrypoint.sh 2. still have issue routing 127.0.0.1 to my browser You should now be able to run command "docker compose -f docker/docker-compose.yml down && docker compose -f docker/docker-compose.yml up --build data-formulator-dev" and see tool on port :5173 — Reply to this email directly, view it on GitHub<[#90 (comment)](https://github.com/microsoft/data-formulator/pull/90#issuecomment-2695342146)>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AWJXEPEQUGQ4VY47DHKEWI32SSUE7AVCNFSM6AAAAABXDTTYCGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJVGM2DEMJUGY>. You are receiving this because you commented.Message ID: ***@***.***>

Updated PR branch as requested

EPS87 commented on docker/docker-compose.yml:
```diff
@@ -0,0 +1,47 @@
+      dockerfile: docker/Dockerfile
+      target: production
+    ports:
+      - "5000:5000"
```
OK let's go with this suggestions
